Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm

```diff
@@ -410,7 +410,7 @@ - (void)_initializeWithConfiguration:(WKWebViewConfiguration *)configuration
 
     _page->setApplicationNameForDesktopUserAgent(configuration._applicationNameForDesktopUserAgent);
 
-    _navigationState = makeUnique<WebKit::NavigationState>(self);
+    _navigationState = WebKit::NavigationState::create(self);
```

Comment:
You're introducing a reference cycle and thus a leak here. WKWebView needs to keep using `makeUnique<WebKit::NavigationState>(self);`. Otherwise, WKWebView refs NavigationState and refine the NavigationState refs the WKWebView again (due to ref counting forwarding).

Reply:
Ah. Thanks! I should be careful when forwarding ref/deref to other object.